src/invoice/api.ts

```diff
@@ -0,0 +1,43 @@
+import {AxiosInstance} from 'axios';
+import {
+    Invoice,
+    ListInvoicesRequest
+} from './interfaces';
+import {Appointment} from "..";
+
+const ENDPOINT = 'invoices';
+
+export class InvoiceApi {
+    constructor(private api: AxiosInstance) {
+    }
+
+  /**
+   * Use this endpoint to query for matching invoices.
+   * @param params - The request object for querying invoices
+   */
+  async listInvoices(
+        params: ListInvoicesRequest
+    ): Promise<Invoice[]> {
+        const urlParams = [];
+
+        Object.keys(params).forEach((key) => {
```

Comment:
can just use `map` to return it straight away. and shouldn't need this if, because it would've already had a key in there?

Reply:
I'm concerned if they pass in a blank field... I don't know how intakeQ will handle that.